src/internal/packager/git/pull.go

```diff
@@ -57,8 +59,38 @@ func (g *Git) Pull(gitURL, targetFolder string) error {
 	repoFolder := fmt.Sprintf("%s-%d", get("repo"), utils.GetCRCHash(gitURL))
 	g.GitPath = path.Join(targetFolder, repoFolder)
 
+	hostPath := get("hostPath")
+
+	if get("proto") == "file://" && !strings.HasPrefix(get("hostPath"), "/") {
+		// If the git URL is a local file path, we need to convert it to an absolute path.
+		// This is because go-git does not support relative paths.
+		//
+		// Example: file://~/dev/podinfo
+		// and
+		// Example: file://../dev/podinfo
+		// Becomes: file:///home/0x1337/dev/podinfo
```

Comment:
This breaks in flux because this isn't where flux transforms these URLs...  This is maybe more obvious with the refactor I did in #809